blocks/product-list-page/product-list-page.js

```diff
@@ -1,321 +1,63 @@
-
-  return html`<div class="sort" disabled=${disabled}>
-    <button disabled=${disabled}>Sort By: ${currentSort.label}</button>
-    <div class="overlay" ref=${sortMenuRef}>
-      <button class="close" onClick=${() => sortMenuRef.current.classList.toggle('active')}>Close</button>
-      <ul>
-        ${options.map((option) => html`<li>
-          <a href="#" class="${currentSort.value === option.value ? 'active' : ''}" onClick=${(e) => {
-  sortMenuRef.current.classList.toggle('active');
-  const [sort, direction = 'asc'] = option.value.split('-');
-  onSort?.(sort, direction);
-  e.preventDefault();
-}}>${option.label}</a>
-        </li>`)}
-      </ul>
-    </div>
-  </div>`;
-}
-
-class ProductListPage extends Component {
-  constructor(props) {
-    const {
-      type = 'category',
-      category,
-    } = props;
-    super();
-
-    this.facetMenuRef = createRef();
-    this.sortMenuRef = createRef();
-
-    const queryParams = parseQueryParams();
-
-    let headline = 'Search Results';
-    let sort = 'relevance';
-    let sortDirection = 'desc';
-    const defaultPageSize = 12;
-    if (type === 'category') {
-      // Get from H1
-      headline = document.querySelector('.default-content-wrapper > h1')?.innerText;
-      sort = 'position';
-      sortDirection = 'asc';
-    }
-
-    /* if (type === 'search') {
-      sampleRUM('search', { source: '.search-input', target: queryParams.searchTerm });
-    } */
+export default async function decorate(block) {
+  const { urlpath, category, type } = readBlockConfig(block);
+  block.textContent = '';
 
-    this.state = {
-      loading: true,
-      pages: 1,
-      currentPage: 1,
-      basePageSize: defaultPageSize,
-      currentPageSize: defaultPageSize,
-      type,
-      category: {
-        name: headline,
-        id: category || null,
+  const widgetProd = '/scripts/widgets/search.js';
+  await loadScript(widgetProd);
+
+  const storeDetails = {
+    environmentId: await getConfigValue('commerce-environment-id'),
+    environmentType: (await getConfigValue('commerce-environment-id')).includes('sandbox') ? 'testing' : '',
+    apiKey: await getConfigValue('commerce-x-api-key'),
+    websiteCode: await getConfigValue('commerce-website-code'),
+    storeCode: await getConfigValue('commerce-store-code'),
+    storeViewCode: await getConfigValue('commerce-store-view-code'),
+    config: {
+      pageSize: 8,
+      perPageConfig: {
+        pageSizeOptions: '12,24,36',
+        defaultPageSizeOption: '24',
       },
-      sort,
-      sortDirection,
-      products: {
-        items: [],
-        total: 0,
+      minQueryLength: '2',
+      currencySymbol: '$',
+      currencyRate: '1',
+      displayOutOfStock: true,
+      allowAllProducts: false,
+      imageCarousel: false,
+      optimizeImages: true,
+      imageBaseWidth: 200,
+      listview: true,
+      displayMode: '', // "" for plp || "PAGE" for category/catalog
```

Comment:
When should a display mode of `PAGE` be used? Currently search.js will fetch all the commerce data but not render anything. Is that the expected behaviour? What is a `category/catalog` view meant to display?

Reply:
In Adobe Commerce Luma there are alternative display modes for category pages which display mostly banners. So most of the time display mode `PAGE` should not be used.

* `displayMode: 'PAGE'`: https://magento2-demo.magebit.com/women.html
* `displayMode: ''`: https://magento2-demo.magebit.com/women/tops-women/hoodies-and-sweatshirts-women.html

Reply:
👍 So the `PAGE` display mode just hasn't been implemented yet in the storefront PLP project?